### ISSUE TITLE
IO Metrics collection level

### ISSUE THREAD

**Issue by DavidWiseman** [trimble-oss/dba-dash#594]
It would be useful to have the option to adjust the IO collection level.  This is currently done at the file level but this might cause issues for instances with a large number of DBs/files.  The cost to store the data is also reduced if there is the option to store the data at database/drive/instance level etc.

Looking into the option of grouping the data on collection there is a potential issue.  If you are using log shipping, the sys.dm_io_virtual_file_stats DMV won't return stats for the DBs during restore.  You can end up with a big spike in IO due to a database dropping in and out of the sys.dm_io_virtual_file_stats DMV.
Possible workarounds: The collection remains at file level but we have the option to limit the granularity of the data stored in the repository.  We could aggregate the data on collection and use something to detect if the files have changed.  Maybe something like `SUM(CHECKSUM(file_handle)) ` - which needs to match when we compare with the staging table.

**Comment by DavidWiseman** [comment]
Another fix would be to maintain the file-level collection granularity for databases in a standby/restoring state, while aggregating for other DBs.    

e.g.
```sql
CASE WHEN @IOCollectionLevel IN(4,5) THEN mf.database_id WHEN d.state=1 OR is_in_standby=1 THEN -d.database_id ELSE -2147483648 END AS database_id, 
CASE WHEN d.state=1 OR d.is_in_standby=1 THEN -vfs.file_id ELSE -2147483648 END AS file_id,

```

Or the collections could be kept as is and the decision made at the repository level.  This fix should be the most reliable but involves collecting and processing more data.

```sql
DECLARE @EngineEdition INT=4
DECLARE @IOCollectionBitMask TINYINT=31
DECLARE @Grp NVARCHAR(MAX)

SELECT @IOCollectionBitMask = @IOCollectionBitMask & CASE WHEN @EngineEdition IN(1,2,3,4) THEN 31 ELSE 3 END

SELECT @Grp = STUFF((SELECT REPLICATE(CHAR(9),3) + ',' + Grp + CHAR(13) + CHAR(10)
FROM (VALUES('(x.InstanceID)',1),
		('(x.DatabaseID,x.FileID,x.Drive)',2),
		('(x.DatabaseID)',4),
		('(x.DatabaseID,x.Drive)',8),
		('(x.Drive)',16)
		) T(Grp,BitMask)
WHERE BitMask & @IOCollectionBitMask >0
FOR XML PATH(''),TYPE).value('.','NVARCHAR(MAX)'),1,4,REPLICATE(CHAR(9),3))

PRINT @Grp
```

**Comment by DavidWiseman** [comment]
Included in 2.40.0 🎉
